pandas/util/_decorators.py

```diff
@@ -212,6 +212,105 @@ def wrapper(*args, **kwargs):
     return _deprecate_kwarg
 
 
+def _format_argument_list(allow_args: Union[List[str], int]):
```

Comment:
Instead of cramming allowed_args into one variable can we have separate parameters for positional and keyword argument allowance? I think a more natural fit that way

Reply:
can you do this?

Reply:
I am not sure I understand you right (not a native English speaker).

Does it mean, for example,
```
_format_argument_list(3, [x, y])
```

should return "except for the arguments x, y and the first 3 positional arguments"? Such warnings should never be produced by `deprecate_nonkeyword_arguments`, but may make this function useful to format some other warning messages. Is it right?